src/platform/plugins/shared/unified_doc_viewer/public/components/observability/attributes/doc_viewer_attributes_overview/attributes_overview.tsx

```diff
@@ -53,32 +72,50 @@ export function AttributesOverview({
 
   const allFields = Object.keys(flattened);
 
-  // it filters out multifields that have a parent, to prevent entries for multifields like this: field, field.keyword, field.whatever
-  const filteredFields = useMemo(
-    () => allFields.filter(shouldShowFieldHandler),
-    [allFields, shouldShowFieldHandler]
-  );
-
   const groupedFields = useMemo(() => {
```

Comment:
Can we move the field grouping functionality into a separate utility file to keep things cleaner and add some unit tests?